src/app/(admin)/admin/_utils/transformSpecialistEditData.js

```diff
@@ -0,0 +1,41 @@
+import { FormatOfWork } from '@prisma/client';
+
+export const transformSpecialistEditData = data => {
+  const therapiesToConnect = data.therapiesIds?.map(id => ({ id })) ?? [];
+  const specializationsToConnect = data.specializationsIds?.map(id => ({ id }));
+  const addressesToConnect = data.addresses?.filter(address => address.id).map(address => ({ id: address.id })) ?? [];
+  const addressesToCreate =
+    data.addresses
+      ?.filter(address => !address.id)
+      .map(address => ({
+        ...address,
+        district: { connect: { id: address.districtId } },
+        districtId: undefined,
+      })) ?? [];
+
+  const unselectedAddresses =
+    data.addressesIds?.filter(addressId => !addressesToConnect.some(address => address.id === addressId)) ?? [];
+  // if formatOfWork is ONLINE, we need to delete all connected addresses
+  const addressesToDelete =
```

Comment:
you sure you need to delete address(not disconnect)?

Reply:
Thanks for comment! Honestly, addresses are only used to belong to specialist. If spec decided to work only online, he doesn't need addresses anymore, so there will be unnecessary information in DB. In my opinion, there wont be a problem for specialist to refill addresses list if he/she wants to work offline again.